src/dataframe/io.jl

```diff
@@ -868,6 +868,73 @@ function readtable(io::IO,
     return df
 end
 
+"""
+Read data from a tabular-file format (CSV, TSV, ...)
+
+```julia
```

Comment:
The convention is to write the signature first, with a four-space ident instead of a ``` block. Here, since there are many options, maybe only write:
```julia
    readtable(filename; [options])
    readtable(io; [options])
```

and detail the arguments only under `Arguments`, with the form `arg::T=default`?  Sounds like a situation to consider in https://github.com/JuliaLang/julia/pull/15136.